Algorithm.CSharp/InteractiveBrokersTieredFeeModelAlgorithm.cs

```diff
@@ -0,0 +1,130 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using QuantConnect.Orders.Fees;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Test algorithm using <see cref="InteractiveBrokersTieredFeeModel"/>
+    /// </summary>
+    public class InteractiveBrokersTieredFeeModelAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _spy, _aig, _bac;
+        private IFeeModel _feeModel = new InteractiveBrokersTieredFeeModel();
+
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 7);   //Set Start Date
+            SetEndDate(2013, 10, 10);     //Set End Date
+            SetCash(1000000000);            //Set Strategy Cash
+
+            // Set the fee model to be shared by all securities to accurately track the volume/value traded to select the correct tiered fee structure.
+            SetSecurityInitializer((security) => security.SetFeeModel(_feeModel));
+
+            _spy = AddEquity("SPY", Resolution.Minute, extendedMarketHours: true).Symbol;
+            _aig = AddEquity("AIG", Resolution.Minute, extendedMarketHours: true).Symbol;
+            _bac = AddEquity("BAC", Resolution.Minute, extendedMarketHours: true).Symbol;
+        }
+
+        public override void OnData(Slice slice)
+        {
+            // Order at different time for various order type to elicit different fee structure.
+            if (slice.Time.Hour == 9 && slice.Time.Minute == 0)
+            {
+                MarketOnOpenOrder(_spy, 30000);
+                MarketOnOpenOrder(_aig, 30000);
+                MarketOnOpenOrder(_bac, 30000);
```

Comment:
Are these orders filled on the next day or right away? If the latest, might as well use MarketOrder? Same for the MarketOnClose ones

Reply:
They should be filled right away at 9:30am, same for market on close at 4pm